CMakeLists.txt

```diff
@@ -272,7 +273,7 @@ link_directories(${PROJECT_BINARY_DIR})
 
 # Load all of our third party directories
 add_subdirectory(thirdparty)
-if (UNIX)
+if (UNIX AND NOT APPLE)
```

Comment:
Please enable upgrading of warnings to errors for apple.

Reply:
Maybe you'll see something I'm not, but I'm getting a warning on the Mac build that doesn't seem... sane. In line 454 of UserManager.cpp, there is a warning about a [[nodiscard]] return value being discarded. But none of those values are being discarded at all, so I don't know what's setting it off.

Reply:
This is the other issue:
![Screenshot_20240123_071403_Chrome](https://github.com/DarkflameUniverse/DarkflameServer/assets/9892985/3f3662d0-984d-4b9b-a85b-c473338317cf)

Reply:
add
```


if (APPLE)
	set_source_files_properties("MD5.cpp" PROPERTIES COMPILE_FLAGS "-Wdeprecated-declarations")
endif()
``` to dCommon/CMakeLists.txt

Reply:
dCommon/CMakeLists.txt, not dGame/CMakeLists.txt

Reply:
Updated